File: java/src/main/java/route/Field.java
package route;

import org.json.JSONObject;

public class Field {

    private Connection _connection;
    private String _route;

    /**
     * Constructor
     *
     * @param connection    class that help to request API
     */
    public Field(Connection connection) {
        this._connection = connection;
        this._route = "fields/";
    }

    /**
     * List all fields
     *
     * @return      Response
     */
    public JSONObject list() {
        return this._connection.request(this._route, "GET");
    }
}

File: csharp/Driver/Field.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RestSharp;

namespace NP6Api
{
    public class Field
    {
        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="connection">class that help to request API</param>
        public Field(Connection connection)
        {
            _connection = connection;
            _route = "fields/";
        }

        /// <summary>
        /// List all fields
        /// </summary>
        /// <returns>Return NResponse with StatusCode and Body</returns>
        public NResponse List()
        {
            return _connection.Request(_route, Method.GET);
        }

        private readonly Connection _connection;
        private string _route;
    }
}
